distributed/http/semaphore/prometheus.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
```

Comment:
If this is the only file here then maybe it doesn't make sense to make this a separate directory.  Maybe it should live in http/scheduler/prometheus.py ?  Or maybe http/scheduler/prometheus_semaphore.py?  Or maybe we make a prometheus/ directory rather than a semaphore directory.  That seems more likely to see further activity in the future.

Although perhaps not, from my perspective semaphores aren't a big deal, but perhaps they are more important to your workloads than I understand.

Reply:
Unfortunately we have some workloads which rely quite strongly on some sort of semaphore since they are quite entangled with a central payload DB and we can't allow the jobs to kill the DB.

So far, we've been using a semaphore in our internal/proprietary job scheduler which is currently in sunset and is being replaced by dask.distributed. The contributions to the semaphore here are essentially the condensation of our past experiences with the matter. Observation and resilience where a big deal for us in the past and we don't want to fall back to old error patterns.

These metrics will most likely wrap up our immediate concerns about the semaphore and I don't expect any major changes or additions incoming after this; assuming nothing breaks :). In particular, I don't see any additions incoming w.r.t the `http` module.

Wrapping up, I like the suggestion of putting this into a `http/scheduler/prometheus` directory since this fits best with the notion of extensions

Reply:
- [ ] move this under `http/scheduler/`